src/averageAges.js

```diff
@@ -13,19 +13,25 @@
  * @param {number} century - optional
  *
  * @return {number}
+ *
+ *
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => century
+    ? Math.ceil(person.died / 100) === century
+    && person.sex === 'm'
+    : person.sex === 'm');
+
+  const ages = men.map(elem => elem.died - elem.born).
+  reduce((a, b) => a + b, 0);
+
+  const average = ages / men.length;
+
+  return average;
```

Comment:
create a separate callback function for this calculations, you repeat the same code the times